src/ORM/Table.php

```diff
@@ -2220,26 +2219,26 @@ public function saveMany(
      * error.
      *
      * @param \Cake\Datasource\ResultSetInterface|array<\Cake\Datasource\EntityInterface> $entities Entities to save.
-     * @param \ArrayAccess|array $options Options used when calling Table::save() for each entity.
+     * @param array $options Options used when calling Table::save() for each entity.
      * @return \Cake\Datasource\ResultSetInterface|array<\Cake\Datasource\EntityInterface> Entities list.
      * @throws \Exception
      * @throws \Cake\ORM\Exception\PersistenceFailedException If an entity couldn't be saved.
      */
-    public function saveManyOrFail(iterable $entities, ArrayAccess|array $options = []): ResultSetInterface|array
+    public function saveManyOrFail(iterable $entities, array $options = []): ResultSetInterface|array
     {
         return $this->_saveMany($entities, $options);
     }
 
     /**
      * @param \Cake\Datasource\ResultSetInterface|array<\Cake\Datasource\EntityInterface> $entities Entities to save.
-     * @param \Cake\ORM\SaveOptionsBuilder|\ArrayAccess|array $options Options used when calling Table::save() for each entity.
+     * @param \Cake\ORM\SaveOptionsBuilder|array $options Options used when calling Table::save() for each entity.
      * @throws \Cake\ORM\Exception\PersistenceFailedException If an entity couldn't be saved.
      * @throws \Exception If an entity couldn't be saved.
      * @return \Cake\Datasource\ResultSetInterface|array<\Cake\Datasource\EntityInterface> Entities list.
      */
     protected function _saveMany(
         ResultSetInterface|array $entities,
-        SaveOptionsBuilder|ArrayAccess|array $options = []
+        SaveOptionsBuilder|array $options = []
```

Comment:
SaveOptionsBuilder is probably not working here. We aren't converting it array properly before wrapping in ArrayObject. This is done correctly for Table::save() above.

Reply:
Fixed to work like Table::save().

Reply:
We should deprecate and remove `SaveOptionsBuilder` in 5.x. It is an incomplete and buggy prototype for replacing option arrays with fluent builders. I suspect it never caught on because it was poorly documented, and only implemented for a single use case.

Reply:
Deprecated in 4.4. Will remove separately after merging.